--- a/soapui/src/main/java/com/eviware/soapui/actions/MockAsWarAction.java
+++ b/soapui/src/main/java/com/eviware/soapui/actions/MockAsWarAction.java
@@ -18,7 +18,6 @@ package com.eviware.soapui.actions;
 
 import com.eviware.soapui.DefaultSoapUICore;
 import com.eviware.soapui.SoapUI;
-import com.eviware.soapui.analytics.Analytics;
 import com.eviware.soapui.analytics.SoapUIActions;
 import com.eviware.soapui.impl.wsdl.WsdlProject;
 import com.eviware.soapui.impl.wsdl.support.HelpUrls;
@@ -85,13 +84,6 @@ public class MockAsWarAction extends Abs
                     project);
             mockAsWar.createMockAsWarArchive();
 
-            if (project.getRestMockServiceCount() > 0) {
-                Analytics.trackAction(SoapUIActions.DEPLOY_REST_MOCK_AS_WAR.getActionName());
-            }
-
-            if (project.getMockServiceCount() > 0) {
-                Analytics.trackAction(SoapUIActions.DEPLOY_SOAP_MOCK_AS_WAR.getActionName());
-            }
         }
     }
 
--- a/soapui/src/main/java/com/eviware/soapui/impl/actions/NewRestProjectAction.java
+++ b/soapui/src/main/java/com/eviware/soapui/impl/actions/NewRestProjectAction.java
@@ -17,7 +17,6 @@
 package com.eviware.soapui.impl.actions;
 
 import com.eviware.soapui.SoapUI;
-import com.eviware.soapui.analytics.Analytics;
 import com.eviware.soapui.analytics.SoapUIActions;
 import com.eviware.soapui.impl.WorkspaceImpl;
 import com.eviware.soapui.impl.wsdl.WsdlProject;
@@ -59,7 +58,6 @@ public class NewRestProjectAction extend
             public void actionPerformed(ActionEvent e) {
                 dialog.setVisible(false);
                 SoapUI.getActionRegistry().getAction(NewWadlProjectAction.SOAPUI_ACTION_ID).perform(SoapUI.getWorkspace(), null);
-                Analytics.trackAction(SoapUIActions.IMPORT_WADL.getActionName());
             }
         });
         while (dialog.show()) {
@@ -72,7 +70,6 @@ public class NewRestProjectAction extend
                 }
                 // If there is no exception or error we break out
 
-                Analytics.trackAction(SoapUIActions.CREATE_REST_PROJECT.getActionName());
                 break;
 
             } catch (Exception ex) {
--- a/soapui/src/main/java/com/eviware/soapui/impl/rest/actions/method/NewRestRequestAction.java
+++ b/soapui/src/main/java/com/eviware/soapui/impl/rest/actions/method/NewRestRequestAction.java
@@ -16,7 +16,6 @@
 
 package com.eviware.soapui.impl.rest.actions.method;
 
-import com.eviware.soapui.analytics.Analytics;
 import com.eviware.soapui.analytics.SoapUIActions;
 import com.eviware.soapui.impl.rest.RestMethod;
 import com.eviware.soapui.impl.rest.RestRequest;
@@ -61,8 +60,6 @@ public class NewRestRequestAction extend
             if (dialog.getBooleanValue(Form.OPENSREQUEST)) {
                 UISupport.showDesktopPanel(request);
             }
-
-            Analytics.trackAction(SoapUIActions.CREATE_REQUEST.getActionName(), "RequestType", "REST");
         }
     }
 
--- a/soapui/src/main/java/com/eviware/soapui/impl/rest/panels/request/AbstractRestRequestDesktopPanel.java
+++ b/soapui/src/main/java/com/eviware/soapui/impl/rest/panels/request/AbstractRestRequestDesktopPanel.java
@@ -16,7 +16,6 @@
 
 package com.eviware.soapui.impl.rest.panels.request;
 
-import com.eviware.soapui.analytics.Analytics;
 import com.eviware.soapui.analytics.SoapUIActions;
 import com.eviware.soapui.impl.rest.RestRequestInterface;
 import com.eviware.soapui.impl.rest.support.RestParamProperty;
@@ -86,7 +85,6 @@ public abstract class AbstractRestReques
 
     @Override
     protected Submit doSubmit() throws SubmitException {
-        Analytics.trackAction(SoapUIActions.RUN_REQUEST_FROM_REQUEST_EDITOR.getActionName(), "Type", "REST");
         return getRequest().submit(new WsdlSubmitContext(getModelItem()), true);
     }
 
--- a/soapui/src/main/java/com/eviware/soapui/impl/wsdl/actions/operation/NewRequestAction.java
+++ b/soapui/src/main/java/com/eviware/soapui/impl/wsdl/actions/operation/NewRequestAction.java
@@ -16,7 +16,6 @@
 
 package com.eviware.soapui.impl.wsdl.actions.operation;
 
-import com.eviware.soapui.analytics.Analytics;
 import com.eviware.soapui.analytics.SoapUIActions;
 import com.eviware.soapui.impl.wsdl.WsdlOperation;
 import com.eviware.soapui.impl.wsdl.WsdlRequest;
@@ -58,7 +57,5 @@ public class NewRequestAction extends Ab
         }
 
         UISupport.showDesktopPanel(newRequest);
-
-        Analytics.trackAction(SoapUIActions.CREATE_REQUEST.getActionName(), "RequestType", "SOAP");
     }
 }
--- a/soapui/src/main/java/com/eviware/soapui/impl/wsdl/actions/project/AddNewMockServiceAction.java
+++ b/soapui/src/main/java/com/eviware/soapui/impl/wsdl/actions/project/AddNewMockServiceAction.java
@@ -16,7 +16,6 @@
 
 package com.eviware.soapui.impl.wsdl.actions.project;
 
-import com.eviware.soapui.analytics.Analytics;
 import com.eviware.soapui.analytics.SoapUIActions;
 import com.eviware.soapui.impl.wsdl.WsdlProject;
 import com.eviware.soapui.impl.wsdl.mock.WsdlMockService;
@@ -41,7 +40,6 @@ public class AddNewMockServiceAction ext
         if (createMockService(target) != null) {
             ;
         }
-        Analytics.trackAction(SoapUIActions.CREATE_SOAP_MOCK.getActionName());
     }
 
     public WsdlMockService createMockService(WsdlProject project) {
--- a/soapui/src/main/java/com/eviware/soapui/impl/wsdl/actions/project/AddNewRestMockServiceAction.java
+++ b/soapui/src/main/java/com/eviware/soapui/impl/wsdl/actions/project/AddNewRestMockServiceAction.java
@@ -15,7 +15,6 @@
 */
 package com.eviware.soapui.impl.wsdl.actions.project;
 
-import com.eviware.soapui.analytics.Analytics;
 import com.eviware.soapui.analytics.SoapUIActions;
 import com.eviware.soapui.impl.wsdl.WsdlProject;
 import com.eviware.soapui.model.mock.MockService;
@@ -31,7 +30,6 @@ public class AddNewRestMockServiceAction
 
     public void perform(WsdlProject target, Object param) {
         if (createRestMockService(target) != null) {
-            Analytics.trackAction(SoapUIActions.CREATE_REST_MOCK.getActionName());
         }
     }
 
--- a/soapui/src/main/java/com/eviware/soapui/impl/wsdl/actions/request/AddRequestToTestCaseAction.java
+++ b/soapui/src/main/java/com/eviware/soapui/impl/wsdl/actions/request/AddRequestToTestCaseAction.java
@@ -17,7 +17,6 @@
 package com.eviware.soapui.impl.wsdl.actions.request;
 
 import com.eviware.soapui.SoapUI;
-import com.eviware.soapui.analytics.Analytics;
 import com.eviware.soapui.analytics.SoapUIActions;
 import com.eviware.soapui.impl.wsdl.WsdlProject;
 import com.eviware.soapui.impl.wsdl.WsdlRequest;
@@ -70,8 +69,6 @@ public class AddRequestToTestCaseAction
         if (testCase != null) {
             addRequest(testCase, request, -1);
         }
-
-        Analytics.trackAction(SoapUIActions.CREATE_TEST_STEP.getActionName());
     }
 
     @Override
--- a/soapui/src/main/java/com/eviware/soapui/impl/wsdl/mock/dispatch/ScriptMockOperationDispatcher.java
+++ b/soapui/src/main/java/com/eviware/soapui/impl/wsdl/mock/dispatch/ScriptMockOperationDispatcher.java
@@ -17,7 +17,6 @@
 package com.eviware.soapui.impl.wsdl.mock.dispatch;
 
 import com.eviware.soapui.SoapUI;
-import com.eviware.soapui.analytics.Analytics;
 import com.eviware.soapui.analytics.SoapUIActions;
 import com.eviware.soapui.impl.support.AbstractMockOperation;
 import com.eviware.soapui.impl.support.actions.ShowOnlineHelpAction;
@@ -210,7 +209,6 @@ public class ScriptMockOperationDispatch
         }
 
         public void actionPerformed(ActionEvent e) {
-            Analytics.trackAction(SoapUIActions.RUN_TEST_STEP.getActionName(), "StepType", "GroovyScript");
 
             MockResult lastMockResult = getMockOperation().getLastMockResult();
             MockRequest mockRequest = lastMockResult == null ? null : lastMockResult.getMockRequest();
--- a/soapui/src/main/java/com/eviware/soapui/impl/wsdl/mock/WsdlMockRunner.java
+++ b/soapui/src/main/java/com/eviware/soapui/impl/wsdl/mock/WsdlMockRunner.java
@@ -17,7 +17,6 @@
 package com.eviware.soapui.impl.wsdl.mock;
 
 import com.eviware.soapui.SoapUI;
-import com.eviware.soapui.analytics.Analytics;
 import com.eviware.soapui.impl.wsdl.WsdlInterface;
 import com.eviware.soapui.impl.wsdl.WsdlOperation;
 import com.eviware.soapui.impl.wsdl.testcase.WsdlTestRunContext;
@@ -206,8 +205,6 @@ public class WsdlMockRunner implements M
         for (MockRunListener listener : mockRunListeners) {
             listener.onMockRunnerStart(this);
         }
-
-        Analytics.trackAction("Start Mock Service, ID: " + getMockService().getStringID());
     }
 
     public void setLogEnabled(boolean logEnabled) {
--- a/soapui/src/main/java/com/eviware/soapui/impl/wsdl/panels/loadtest/JLoadTestAssertionsTable.java
+++ b/soapui/src/main/java/com/eviware/soapui/impl/wsdl/panels/loadtest/JLoadTestAssertionsTable.java
@@ -16,7 +16,6 @@
 
 package com.eviware.soapui.impl.wsdl.panels.loadtest;
 
-import com.eviware.soapui.analytics.Analytics;
 import com.eviware.soapui.analytics.SoapUIActions;
 import com.eviware.soapui.impl.support.actions.ShowOnlineHelpAction;
 import com.eviware.soapui.impl.wsdl.loadtest.LoadTestAssertion;
@@ -278,7 +277,6 @@ public class JLoadTestAssertionsTable ex
             String type = (String) UISupport.prompt("Select assertion type to add", "Add Assertion", types);
             if (type != null) {
                 loadTest.addAssertion(type, LoadTestAssertion.ANY_TEST_STEP, true);
-                Analytics.trackAction(SoapUIActions.ADD_LOAD_TEST_ASSERTION.getActionName());
             }
         }
     }
--- a/soapui/src/main/java/com/eviware/soapui/impl/wsdl/panels/loadtest/WsdlLoadTestDesktopPanel.java
+++ b/soapui/src/main/java/com/eviware/soapui/impl/wsdl/panels/loadtest/WsdlLoadTestDesktopPanel.java
@@ -17,7 +17,6 @@
 package com.eviware.soapui.impl.wsdl.panels.loadtest;
 
 import com.eviware.soapui.SoapUI;
-import com.eviware.soapui.analytics.Analytics;
 import com.eviware.soapui.analytics.SoapUIActions;
 import com.eviware.soapui.config.LoadTestLimitTypesConfig;
 import com.eviware.soapui.impl.support.actions.ShowOnlineHelpAction;
@@ -393,7 +392,6 @@ public class WsdlLoadTestDesktopPanel ex
 
             runButton.setEnabled(false);
             runner = loadtest.run();
-            Analytics.trackAction(SoapUIActions.RUN_LOAD_TEST.getActionName());
         }
     }
 
--- a/soapui/src/main/java/com/eviware/soapui/impl/wsdl/panels/testcase/TestOnDemandPanel.java
+++ b/soapui/src/main/java/com/eviware/soapui/impl/wsdl/panels/testcase/TestOnDemandPanel.java
@@ -18,7 +18,6 @@ package com.eviware.soapui.impl.wsdl.pan
 
 import com.eviware.soapui.SoapUI;
 import com.eviware.soapui.SoapUISystemProperties;
-import com.eviware.soapui.analytics.Analytics;
 import com.eviware.soapui.analytics.SoapUIActions;
 import com.eviware.soapui.impl.support.actions.ShowOnlineHelpAction;
 import com.eviware.soapui.impl.wsdl.support.HelpUrls;
@@ -247,8 +246,6 @@ public class TestOnDemandPanel extends J
                 if (!Strings.isNullOrEmpty(redirectUrl)) {
                     openURLSafely(redirectUrl);
                 }
-
-                Analytics.trackAction(SoapUIActions.TEST_ON_DEMAND.getActionName());
             }
         }
     }
--- a/soapui/src/main/java/com/eviware/soapui/impl/wsdl/panels/teststeps/amf/AMFRequestTestStepDesktopPanel.java
+++ b/soapui/src/main/java/com/eviware/soapui/impl/wsdl/panels/teststeps/amf/AMFRequestTestStepDesktopPanel.java
@@ -17,7 +17,6 @@
 package com.eviware.soapui.impl.wsdl.panels.teststeps.amf;
 
 import com.eviware.soapui.SoapUI;
-import com.eviware.soapui.analytics.Analytics;
 import com.eviware.soapui.analytics.SoapUIActions;
 import com.eviware.soapui.config.AMFRequestTestStepConfig;
 import com.eviware.soapui.impl.support.actions.ShowOnlineHelpAction;
@@ -720,8 +719,6 @@ public class AMFRequestTestStepDesktopPa
             throw new SubmitException("AMF request is not initialised properly !");
         }
 
-        Analytics.trackAction(SoapUIActions.RUN_TEST_STEP.getActionName(), "StepType", "AMF");
-
         return amfRequestTestStep.getAMFRequest().submit(submitContext, true);
     }
 
--- a/soapui/src/main/java/com/eviware/soapui/impl/wsdl/panels/teststeps/HttpTestRequestDesktopPanel.java
+++ b/soapui/src/main/java/com/eviware/soapui/impl/wsdl/panels/teststeps/HttpTestRequestDesktopPanel.java
@@ -17,7 +17,6 @@
 package com.eviware.soapui.impl.wsdl.panels.teststeps;
 
 import com.eviware.soapui.SoapUI;
-import com.eviware.soapui.analytics.Analytics;
 import com.eviware.soapui.analytics.SoapUIActions;
 import com.eviware.soapui.impl.rest.RestRequestInterface;
 import com.eviware.soapui.impl.support.AbstractHttpRequest;
@@ -324,8 +323,6 @@ public class HttpTestRequestDesktopPanel
 
     @Override
     protected Submit doSubmit() throws SubmitException {
-        Analytics.trackAction(SoapUIActions.RUN_TEST_STEP.getActionName(), "StepType", "HTTP");
-
         return getRequest().submit(new WsdlTestRunContext(getModelItem()), true);
     }
 
--- a/soapui/src/main/java/com/eviware/soapui/impl/wsdl/panels/teststeps/JdbcRequestTestStepDesktopPanel.java
+++ b/soapui/src/main/java/com/eviware/soapui/impl/wsdl/panels/teststeps/JdbcRequestTestStepDesktopPanel.java
@@ -17,7 +17,6 @@
 package com.eviware.soapui.impl.wsdl.panels.teststeps;
 
 import com.eviware.soapui.SoapUI;
-import com.eviware.soapui.analytics.Analytics;
 import com.eviware.soapui.analytics.SoapUIActions;
 import com.eviware.soapui.config.JdbcRequestTestStepConfig;
 import com.eviware.soapui.impl.support.actions.ShowOnlineHelpAction;
@@ -687,8 +686,6 @@ public class JdbcRequestTestStepDesktopP
     }
 
     protected Submit doSubmit() throws SubmitException {
-        Analytics.trackAction(SoapUIActions.RUN_TEST_STEP.getActionName(), "StepType", "JDBC");
-
         return jdbcRequestTestStep.getJdbcRequest().submit(new WsdlTestRunContext(getModelItem()), true);
     }
 
--- a/soapui/src/main/java/com/eviware/soapui/impl/wsdl/panels/teststeps/RestTestRequestDesktopPanel.java
+++ b/soapui/src/main/java/com/eviware/soapui/impl/wsdl/panels/teststeps/RestTestRequestDesktopPanel.java
@@ -17,7 +17,6 @@
 package com.eviware.soapui.impl.wsdl.panels.teststeps;
 
 import com.eviware.soapui.SoapUI;
-import com.eviware.soapui.analytics.Analytics;
 import com.eviware.soapui.analytics.SoapUIActions;
 import com.eviware.soapui.impl.rest.RestMethod;
 import com.eviware.soapui.impl.rest.RestResource;
@@ -230,8 +229,6 @@ public class RestTestRequestDesktopPanel
     }
 
     protected Submit doSubmit() throws SubmitException {
-        Analytics.trackAction(SoapUIActions.RUN_TEST_STEP.getActionName(), "StepType", "REST");
-
         return getRequest().submit(new WsdlTestRunContext(getModelItem()), true);
     }
 
--- a/soapui/src/main/java/com/eviware/soapui/impl/wsdl/panels/teststeps/WsdlRunTestCaseStepDesktopPanel.java
+++ b/soapui/src/main/java/com/eviware/soapui/impl/wsdl/panels/teststeps/WsdlRunTestCaseStepDesktopPanel.java
@@ -17,7 +17,6 @@
 package com.eviware.soapui.impl.wsdl.panels.teststeps;
 
 import com.eviware.soapui.SoapUI;
-import com.eviware.soapui.analytics.Analytics;
 import com.eviware.soapui.analytics.SoapUIActions;
 import com.eviware.soapui.config.RunTestCaseRunModeTypeConfig;
 import com.eviware.soapui.impl.support.actions.ShowOnlineHelpAction;
@@ -199,7 +198,6 @@ public class WsdlRunTestCaseStepDesktopP
         }
 
         public void actionPerformed(ActionEvent e) {
-            Analytics.trackAction(SoapUIActions.RUN_TEST_STEP.getActionName(), "StepType", "RunTestCase");
 
             runAction.setEnabled(false);
             cancelAction.setEnabled(true);
--- a/soapui/src/main/java/com/eviware/soapui/impl/wsdl/panels/teststeps/WsdlTestRequestDesktopPanel.java
+++ b/soapui/src/main/java/com/eviware/soapui/impl/wsdl/panels/teststeps/WsdlTestRequestDesktopPanel.java
@@ -17,7 +17,6 @@
 package com.eviware.soapui.impl.wsdl.panels.teststeps;
 
 import com.eviware.soapui.SoapUI;
-import com.eviware.soapui.analytics.Analytics;
 import com.eviware.soapui.analytics.SoapUIActions;
 import com.eviware.soapui.impl.support.components.ModelItemXmlEditor;
 import com.eviware.soapui.impl.wsdl.panels.request.AbstractWsdlRequestDesktopPanel;
@@ -184,8 +183,6 @@ public class WsdlTestRequestDesktopPanel
     }
 
     protected Submit doSubmit() throws SubmitException {
-        Analytics.trackAction(SoapUIActions.RUN_TEST_STEP.getActionName(), "StepType", "SOAP");
-
         return getRequest().submit(new WsdlTestRunContext(getModelItem()), true);
     }
 
--- a/soapui/src/main/java/com/eviware/soapui/impl/wsdl/panels/testsuite/WsdlTestSuiteDesktopPanel.java
+++ b/soapui/src/main/java/com/eviware/soapui/impl/wsdl/panels/testsuite/WsdlTestSuiteDesktopPanel.java
@@ -17,7 +17,6 @@
 package com.eviware.soapui.impl.wsdl.panels.testsuite;
 
 import com.eviware.soapui.SoapUI;
-import com.eviware.soapui.analytics.Analytics;
 import com.eviware.soapui.analytics.SoapUIActions;
 import com.eviware.soapui.impl.support.actions.ShowOnlineHelpAction;
 import com.eviware.soapui.impl.wsdl.WsdlTestSuite;
@@ -333,7 +332,6 @@ public class WsdlTestSuiteDesktopPanel e
 
         public void actionPerformed(ActionEvent e) {
             runTestSuite();
-            Analytics.trackAction(SoapUIActions.RUN_TEST_SUITE.getActionName(), "RunType", getModelItem().getRunType().name());
         }
     }
 
--- a/soapui/src/main/java/com/eviware/soapui/impl/wsdl/testcase/WsdlTestCase.java
+++ b/soapui/src/main/java/com/eviware/soapui/impl/wsdl/testcase/WsdlTestCase.java
@@ -17,7 +17,6 @@
 package com.eviware.soapui.impl.wsdl.testcase;
 
 import com.eviware.soapui.SoapUI;
-import com.eviware.soapui.analytics.Analytics;
 import com.eviware.soapui.config.LoadTestConfig;
 import com.eviware.soapui.config.SecurityTestConfig;
 import com.eviware.soapui.config.TestCaseConfig;
@@ -510,8 +509,6 @@ public class WsdlTestCase extends Abstra
 
         notifyPropertyChanged("testSteps", null, testStep);
 
-        Analytics.trackAction("AddRequestToTestCase", "Type", testStep.getClass().getSimpleName());
-
         return testStep;
     }
 
--- a/soapui/src/main/java/com/eviware/soapui/impl/wsdl/WsdlProject.java
+++ b/soapui/src/main/java/com/eviware/soapui/impl/wsdl/WsdlProject.java
@@ -17,7 +17,6 @@
 package com.eviware.soapui.impl.wsdl;
 
 import com.eviware.soapui.SoapUI;
-import com.eviware.soapui.analytics.Analytics;
 import com.eviware.soapui.analytics.SoapUIActions;
 import com.eviware.soapui.config.InterfaceConfig;
 import com.eviware.soapui.config.MockServiceConfig;
@@ -1216,8 +1215,6 @@ public class WsdlProject extends Abstrac
         addWsdlMockService(mockService);
         fireMockServiceAdded(mockService);
 
-        Analytics.trackAction(SoapUIActions.CREATE_SOAP_MOCK.getActionName());
-
         return mockService;
     }
 
@@ -1231,8 +1228,6 @@ public class WsdlProject extends Abstrac
         addRestMockService(mockService);
         fireMockServiceAdded(mockService);
 
-        Analytics.trackAction(SoapUIActions.CREATE_REST_MOCK.getActionName());
-
         return mockService;
     }
 
--- a/soapui/src/main/java/com/eviware/soapui/impl/WsdlInterfaceFactory.java
+++ b/soapui/src/main/java/com/eviware/soapui/impl/WsdlInterfaceFactory.java
@@ -17,7 +17,6 @@
 package com.eviware.soapui.impl;
 
 import com.eviware.soapui.SoapUI;
-import com.eviware.soapui.analytics.Analytics;
 import com.eviware.soapui.analytics.SoapUIActions;
 import com.eviware.soapui.config.InterfaceConfig;
 import com.eviware.soapui.config.WsdlInterfaceConfig;
@@ -103,8 +102,6 @@ public class WsdlInterfaceFactory implem
             throw new SoapUIException("Error creating requests", e);
         }
 
-        Analytics.trackAction(SoapUIActions.IMPORT_WSDL.getActionName());
-
         return result;
     }
 
--- a/soapui/src/main/java/com/eviware/soapui/security/panels/SecurityTestDesktopPanel.java
+++ b/soapui/src/main/java/com/eviware/soapui/security/panels/SecurityTestDesktopPanel.java
@@ -17,7 +17,6 @@
 package com.eviware.soapui.security.panels;
 
 import com.eviware.soapui.SoapUI;
-import com.eviware.soapui.analytics.Analytics;
 import com.eviware.soapui.analytics.SoapUIActions;
 import com.eviware.soapui.impl.support.actions.ShowOnlineHelpAction;
 import com.eviware.soapui.impl.wsdl.panels.support.MockSecurityTestRunner;
@@ -417,7 +416,6 @@ public class SecurityTestDesktopPanel ex
                 }
             }
             if (shouldRun) {
-                Analytics.trackAction(SoapUIActions.RUN_SECURITY_TEST.getActionName());
                 runSecurityTest();
             } else {
                 UISupport.showInfoMessage("No Security Scans available to run.", "Security Test Warning");
--- a/soapui/src/main/java/com/eviware/soapui/SoapUI.java
+++ b/soapui/src/main/java/com/eviware/soapui/SoapUI.java
@@ -22,7 +22,6 @@ import com.eviware.soapui.actions.SoapUI
 import com.eviware.soapui.actions.StartHermesJMSButtonAction;
 import com.eviware.soapui.actions.SumbitUserInfoAction;
 import com.eviware.soapui.actions.SwitchDesktopPanelAction;
-import com.eviware.soapui.analytics.Analytics;
 import com.eviware.soapui.analytics.AnalyticsHelper;
 import com.eviware.soapui.impl.WorkspaceImpl;
 import com.eviware.soapui.impl.actions.ImportWsdlProjectAction;
@@ -714,7 +713,6 @@ public class SoapUI {
                 }
 
                 if (isCommandLine()) {
-                    Analytics.trackAction("CmdLine");
                 }
             } catch (Exception e) {
                 e.printStackTrace();
@@ -854,12 +852,9 @@ public class SoapUI {
         soapUICore = core;
 
         AnalyticsHelper.InitializeAnalytics();
-        Analytics.trackSessionStart();
+
         boolean isDebug = java.lang.management.ManagementFactory.getRuntimeMXBean().
                 getInputArguments().toString().indexOf("-agentlib:jdwp") > 0;
-        if (isDebug) {
-            Analytics.trackAction("DebuggingMode");
-        }
 
         SoapUI soapUI = new SoapUI();
         Workspace workspace = null;
@@ -1063,17 +1058,13 @@ public class SoapUI {
                 SoapUI.logError(e1);
             }
 
-            Analytics.trackAction("Exit");
         } else {
             if (!UISupport.confirm("Exit SoapUI without saving?", "Question")) {
                 saveOnExit = true;
                 return false;
             }
-            Analytics.trackAction("ExitWithoutSave");
         }
 
-        Analytics.trackSessionStop();
-
         shutdown();
 
         return true;
--- a/soapui/src/main/java/com/eviware/soapui/support/resolver/CreateNewEmptyTestCase.java
+++ b/soapui/src/main/java/com/eviware/soapui/support/resolver/CreateNewEmptyTestCase.java
@@ -16,7 +16,6 @@
 
 package com.eviware.soapui.support.resolver;
 
-import com.eviware.soapui.analytics.Analytics;
 import com.eviware.soapui.analytics.SoapUIActions;
 import com.eviware.soapui.impl.wsdl.testcase.WsdlTestCase;
 import com.eviware.soapui.impl.wsdl.teststeps.WsdlRunTestCaseTestStep;
@@ -53,8 +52,6 @@ public class CreateNewEmptyTestCase impl
         testStep.setTargetTestCase(tCase);
         resolved = true;
 
-        Analytics.trackAction(SoapUIActions.CREATE_TEST_CASE.getActionName());
-
         return resolved;
     }
 
--- a/soapui/src/main/java/com/eviware/soapui/support/UISupport.java
+++ b/soapui/src/main/java/com/eviware/soapui/support/UISupport.java
@@ -18,7 +18,6 @@ package com.eviware.soapui.support;
 
 import com.eviware.soapui.SoapUI;
 import com.eviware.soapui.SwingPluginSoapUICore;
-import com.eviware.soapui.analytics.Analytics;
 import com.eviware.soapui.impl.wsdl.actions.iface.tools.support.ToolHost;
 import com.eviware.soapui.model.ModelItem;
 import com.eviware.soapui.model.settings.Settings;
@@ -254,7 +253,6 @@ public class UISupport {
     }
 
     public static void showErrorMessage(final String message) {
-        Analytics.trackError(new Throwable(message));
         if (message != null && message.length() > EXTENDED_ERROR_MESSAGE_THRESHOLD) {
             dialogs.showExtendedInfo("Error", "An error occurred", message, null);
         } else {
@@ -596,7 +594,6 @@ public class UISupport {
 
     public static void showErrorMessage(Throwable ex) {
         SoapUI.logError(ex);
-        Analytics.trackError(ex);
 
         if (ex.toString().length() > 100) {
             dialogs.showExtendedInfo("Error", "An error of type " + ex.getClass().getSimpleName() + " occured.",
--- a/soapui/src/main/java/com/eviware/soapui/tools/SoapUITestCaseRunner.java
+++ b/soapui/src/main/java/com/eviware/soapui/tools/SoapUITestCaseRunner.java
@@ -18,7 +18,6 @@ package com.eviware.soapui.tools;
 
 import com.eviware.soapui.analytics.AnalyticsHelper;
 import com.eviware.soapui.SoapUI;
-import com.eviware.soapui.analytics.Analytics;
 import com.eviware.soapui.impl.wsdl.WsdlProject;
 import com.eviware.soapui.impl.wsdl.WsdlTestSuite;
 import com.eviware.soapui.impl.wsdl.testcase.WsdlProjectRunner;
@@ -116,7 +115,6 @@ public class SoapUITestCaseRunner extend
 
     public static void main(String[] args) throws Exception {
         AnalyticsHelper.InitializeAnalytics();
-        Analytics.trackSessionStart();
         System.exit(new SoapUITestCaseRunner().runFromCommandLine(args));
     }
 
--- a/soapui/src/main/java/com/eviware/soapui/impl/wsdl/actions/iface/tools/support/SwingToolHost.java
+++ b/soapui/src/main/java/com/eviware/soapui/impl/wsdl/actions/iface/tools/support/SwingToolHost.java
@@ -16,7 +16,6 @@
 
 package com.eviware.soapui.impl.wsdl.actions.iface.tools.support;
 
-import com.eviware.soapui.analytics.Analytics;
 import com.eviware.soapui.analytics.SoapUIActions;
 import com.eviware.soapui.model.ModelItem;
 import com.eviware.soapui.support.UISupport;
@@ -45,7 +44,6 @@ public class SwingToolHost implements To
             UISupport.showErrorMessage(ex);
             throw ex;
         } finally {
-            Analytics.trackAction(SoapUIActions.RUN_TOOL.getActionName(), "Tool", runner.getName());
             if (processDialog != null) {
                 processDialog.setVisible(false);
             }
--- a/soapui/src/main/java/com/eviware/soapui/impl/wsdl/panels/request/AbstractWsdlRequestDesktopPanel.java
+++ b/soapui/src/main/java/com/eviware/soapui/impl/wsdl/panels/request/AbstractWsdlRequestDesktopPanel.java
@@ -16,7 +16,6 @@
 
 package com.eviware.soapui.impl.wsdl.panels.request;
 
-import com.eviware.soapui.analytics.Analytics;
 import com.eviware.soapui.analytics.SoapUIActions;
 import com.eviware.soapui.impl.support.components.ModelItemXmlEditor;
 import com.eviware.soapui.impl.support.components.RequestXmlDocument;
@@ -137,7 +136,6 @@ public abstract class AbstractWsdlReques
     }
 
     protected Submit doSubmit() throws SubmitException {
-        Analytics.trackAction(SoapUIActions.RUN_REQUEST_FROM_REQUEST_EDITOR.getActionName(), "Type", "SOAP");
         return getRequest().submit(new WsdlSubmitContext(getModelItem()), true);
     }
 
--- a/soapui/src/main/java/com/eviware/soapui/impl/wsdl/panels/teststeps/PropertyTransfersDesktopPanel.java
+++ b/soapui/src/main/java/com/eviware/soapui/impl/wsdl/panels/teststeps/PropertyTransfersDesktopPanel.java
@@ -16,7 +16,6 @@
 
 package com.eviware.soapui.impl.wsdl.panels.teststeps;
 
-import com.eviware.soapui.analytics.Analytics;
 import com.eviware.soapui.analytics.SoapUIActions;
 import com.eviware.soapui.impl.support.actions.ShowOnlineHelpAction;
 import com.eviware.soapui.impl.wsdl.panels.support.MockTestRunContext;
@@ -977,8 +976,6 @@ public class PropertyTransfersDesktopPan
 
         public void actionPerformed(ActionEvent e) {
 
-            Analytics.trackAction(SoapUIActions.ADD_PROPERTY_TRASNFER_IN_PROPERTY_TRANSFER_TEST_STEP.getActionName());
-
             String name = UISupport.prompt("Specify name for value transfer", "Add Transfer", "");
             if (name == null || name.trim().length() == 0) {
                 return;
@@ -1167,8 +1164,6 @@ public class PropertyTransfersDesktopPan
                 return;
             }
 
-            Analytics.trackAction(SoapUIActions.RUN_TEST_STEP.getActionName(), "RequestType", "PropertyTransfer");
-
             MockTestRunner mockRunner = new MockTestRunner(transferStep.getTestCase());
             MockTestRunContext context = new MockTestRunContext(mockRunner, transferStep);
             PropertyTransferResult result = (PropertyTransferResult) transferStep.run(mockRunner, context,
--- a/soapui/src/main/java/com/eviware/soapui/impl/wsdl/panels/project/WsdlProjectTestSuitesTabPanel.java
+++ b/soapui/src/main/java/com/eviware/soapui/impl/wsdl/panels/project/WsdlProjectTestSuitesTabPanel.java
@@ -16,7 +16,6 @@
 
 package com.eviware.soapui.impl.wsdl.panels.project;
 
-import com.eviware.soapui.analytics.Analytics;
 import com.eviware.soapui.analytics.SoapUIActions;
 import com.eviware.soapui.impl.support.actions.ShowOnlineHelpAction;
 import com.eviware.soapui.impl.wsdl.WsdlProject;
@@ -276,7 +275,6 @@ public class WsdlProjectTestSuitesTabPan
 
         public void actionPerformed(ActionEvent e) {
             runProject();
-            Analytics.trackAction(SoapUIActions.RUN_PROJECT.getActionName());
         }
     }
 
--- a/soapui/src/main/java/com/eviware/soapui/impl/wsdl/panels/mock/WsdlMockServiceDesktopPanel.java
+++ b/soapui/src/main/java/com/eviware/soapui/impl/wsdl/panels/mock/WsdlMockServiceDesktopPanel.java
@@ -17,7 +17,6 @@
 package com.eviware.soapui.impl.wsdl.panels.mock;
 
 import com.eviware.soapui.SoapUI;
-import com.eviware.soapui.analytics.Analytics;
 import com.eviware.soapui.analytics.SoapUIActions;
 import com.eviware.soapui.impl.rest.actions.mock.AddEmptyRestMockResourceAction;
 import com.eviware.soapui.impl.rest.actions.mock.RestMockServiceOptionsAction;
@@ -545,9 +544,9 @@ public class WsdlMockServiceDesktopPanel
         public void actionPerformed(ActionEvent arg0) {
 
             if (getModelItem() instanceof WsdlMockService) {
-                Analytics.trackAction(SoapUIActions.START_SOAP_MOCK.getActionName());
+
             } else if (getModelItem() instanceof RestMockService) {
-                Analytics.trackAction(SoapUIActions.START_REST_MOCK.getActionName());
+
             }
 
             startMockService();
--- a/soapui/src/main/java/com/eviware/soapui/impl/actions/NewWsdlProjectAction.java
+++ b/soapui/src/main/java/com/eviware/soapui/impl/actions/NewWsdlProjectAction.java
@@ -16,7 +16,6 @@
 
 package com.eviware.soapui.impl.actions;
 
-import com.eviware.soapui.analytics.Analytics;
 import com.eviware.soapui.analytics.SoapUIActions;
 import com.eviware.soapui.impl.WorkspaceImpl;
 import com.eviware.soapui.impl.WsdlInterfaceFactory;
@@ -121,7 +120,6 @@ public class NewWsdlProjectAction extend
 
                             importWsdl(project, url);
                         }
-                        Analytics.trackAction(SoapUIActions.CREATE_SOAP_PROJECT.getActionName());
 
                         break;
                     }
--- a/soapui/src/main/java/com/eviware/soapui/impl/wsdl/actions/project/AddWadlAction.java
+++ b/soapui/src/main/java/com/eviware/soapui/impl/wsdl/actions/project/AddWadlAction.java
@@ -16,7 +16,6 @@
 
 package com.eviware.soapui.impl.wsdl.actions.project;
 
-import com.eviware.soapui.analytics.Analytics;
 import com.eviware.soapui.analytics.SoapUIActions;
 import com.eviware.soapui.impl.rest.RestService;
 import com.eviware.soapui.impl.rest.RestServiceFactory;
@@ -108,8 +107,6 @@ public class AddWadlAction extends Abstr
             UISupport.showErrorMessage(e);
         }
 
-        Analytics.trackAction(SoapUIActions.IMPORT_WADL.getActionName());
-
         return restService;
     }
 
--- a/soapui/src/main/java/com/eviware/soapui/impl/wsdl/actions/testcase/AddNewLoadTestAction.java
+++ b/soapui/src/main/java/com/eviware/soapui/impl/wsdl/actions/testcase/AddNewLoadTestAction.java
@@ -16,7 +16,6 @@
 
 package com.eviware.soapui.impl.wsdl.actions.testcase;
 
-import com.eviware.soapui.analytics.Analytics;
 import com.eviware.soapui.analytics.SoapUIActions;
 import com.eviware.soapui.impl.wsdl.loadtest.WsdlLoadTest;
 import com.eviware.soapui.impl.wsdl.testcase.WsdlTestCase;
@@ -46,6 +45,5 @@ public class AddNewLoadTestAction extend
 
         WsdlLoadTest loadTest = testCase.addNewLoadTest(name);
         UISupport.selectAndShow(loadTest);
-        Analytics.trackAction(SoapUIActions.CREATE_LOAD_TEST.getActionName());
     }
 }
--- a/soapui/src/main/java/com/eviware/soapui/impl/wsdl/actions/testcase/AddNewSecurityTestAction.java
+++ b/soapui/src/main/java/com/eviware/soapui/impl/wsdl/actions/testcase/AddNewSecurityTestAction.java
@@ -16,7 +16,6 @@
 
 package com.eviware.soapui.impl.wsdl.actions.testcase;
 
-import com.eviware.soapui.analytics.Analytics;
 import com.eviware.soapui.analytics.SoapUIActions;
 import com.eviware.soapui.impl.wsdl.testcase.WsdlTestCase;
 import com.eviware.soapui.security.SecurityTest;
@@ -51,6 +50,5 @@ public class AddNewSecurityTestAction ex
 
         SecurityTest securityTest = testCase.addNewSecurityTest(name);
         UISupport.selectAndShow(securityTest);
-        Analytics.trackAction(SoapUIActions.CREATE_SECURITY_TEST.getActionName());
     }
 }
--- a/soapui/src/main/java/com/eviware/soapui/impl/wsdl/actions/testsuite/AddNewTestCaseAction.java
+++ b/soapui/src/main/java/com/eviware/soapui/impl/wsdl/actions/testsuite/AddNewTestCaseAction.java
@@ -16,7 +16,6 @@
 
 package com.eviware.soapui.impl.wsdl.actions.testsuite;
 
-import com.eviware.soapui.analytics.Analytics;
 import com.eviware.soapui.analytics.SoapUIActions;
 import com.eviware.soapui.impl.wsdl.WsdlTestSuite;
 import com.eviware.soapui.impl.wsdl.testcase.WsdlTestCase;
@@ -52,7 +51,6 @@ public class AddNewTestCaseAction extend
         }
 
         WsdlTestCase testCase = testSuite.addNewTestCase(name);
-        Analytics.trackAction(SoapUIActions.CREATE_TEST_CASE.getActionName());
         UISupport.showDesktopPanel(testCase);
     }
 }
--- a/soapui/src/main/java/com/eviware/soapui/impl/wsdl/panels/testcase/WsdlTestCaseDesktopPanel.java
+++ b/soapui/src/main/java/com/eviware/soapui/impl/wsdl/panels/testcase/WsdlTestCaseDesktopPanel.java
@@ -17,7 +17,6 @@
 package com.eviware.soapui.impl.wsdl.panels.testcase;
 
 import com.eviware.soapui.SoapUI;
-import com.eviware.soapui.analytics.Analytics;
 import com.eviware.soapui.analytics.SoapUIActions;
 import com.eviware.soapui.config.TestStepConfig;
 import com.eviware.soapui.impl.support.actions.ShowOnlineHelpAction;
@@ -532,7 +531,6 @@ public class WsdlTestCaseDesktopPanel ex
         public void actionPerformed(ActionEvent e) {
             canceled = false;
             runTestCase();
-            Analytics.trackAction(SoapUIActions.RUN_TEST_CASE.getActionName());
         }
     }
 
--- a/soapui/src/main/java/com/eviware/soapui/tools/SoapUIToolRunner.java
+++ b/soapui/src/main/java/com/eviware/soapui/tools/SoapUIToolRunner.java
@@ -17,7 +17,6 @@
 package com.eviware.soapui.tools;
 
 import com.eviware.soapui.SoapUI;
-import com.eviware.soapui.analytics.Analytics;
 import com.eviware.soapui.analytics.SoapUIActions;
 import com.eviware.soapui.impl.wsdl.WsdlProject;
 import com.eviware.soapui.impl.wsdl.WsdlProjectFactory;
@@ -148,8 +147,6 @@ public class SoapUIToolRunner extends Ab
     public void run(ToolRunner runner) throws Exception {
         status = RunnerStatus.RUNNING;
         runner.setContext(this);
-        Analytics.trackAction(SoapUIActions.RUN_TOOL_FROM_COMMAND_LINE.getActionName(), "Tool", runner.getName());
-
         runner.run();
     }
 
