--- a/soapui/src/main/java/com/eviware/soapui/impl/wsdl/panels/teststeps/amf/SoapUIAMFConnection.java
+++ b/soapui/src/main/java/com/eviware/soapui/impl/wsdl/panels/teststeps/amf/SoapUIAMFConnection.java
@@ -50,7 +50,6 @@
 import flex.messaging.io.amf.client.AMFHeaderProcessor;
 import flex.messaging.io.amf.client.exceptions.ClientStatusException;
 import flex.messaging.io.amf.client.exceptions.ServerStatusException;
-import flex.messaging.io.amf.client.exceptions.ServerStatusException.HttpResponseInfo;
 
 /**
  * AMFConnection derivate using HttpClient instead of UrlConnection
@@ -302,6 +301,7 @@
      *
      * @return The HTTP response info for the server status exception.
      */
+/*
     protected HttpResponseInfo generateHttpResponseInfo() {
         HttpResponseInfo httpResponseInfo = null;
         try {
@@ -315,7 +315,7 @@
         }
         return httpResponseInfo;
     }
-
+*/
     /**
      * Generates and returns the response URI.
      *
