--- a/soapui/src/main/java/com/eviware/soapui/SoapUI.java
+++ b/soapui/src/main/java/com/eviware/soapui/SoapUI.java
@@ -705,10 +705,6 @@
                     });
                 }
 
-                if (isFirstLaunch) {
-                    Tools.openURL(HelpUrls.SOAPUI_WELCOME_PAGE);
-                }
-
                 if (isCommandLine()) {
                 }
             } catch (Exception e) {
