--- a/soapui/src/main/java/com/eviware/soapui/support/UISupport.java
+++ b/soapui/src/main/java/com/eviware/soapui/support/UISupport.java
@@ -40,7 +40,6 @@
 import com.eviware.x.impl.swing.SwingFileDialogs;
 import com.jgoodies.looks.HeaderStyle;
 import com.jgoodies.looks.Options;
-import org.syntax.jedit.InputHandler;
 
 import javax.swing.AbstractAction;
 import javax.swing.Action;
@@ -694,16 +693,7 @@
      */
 
     public static KeyStroke getKeyStroke(String keyStroke) {
-        try {
-            if (InputHandler.getMenuShortcutKeyMask() == Event.META_MASK) {
-                keyStroke = keyStroke.replaceAll("menu", "meta");
-            } else {
-                keyStroke = keyStroke.replaceAll("menu", "ctrl");
-            }
-        } catch (Throwable e) {
-            keyStroke = keyStroke.replaceAll("menu", "ctrl");
-        }
-
+        keyStroke = keyStroke.replaceAll("menu", "ctrl");
         return KeyStroke.getKeyStroke(keyStroke);
     }
 
@@ -1005,4 +995,4 @@
         }
         return null;
     }
-}
\ No newline at end of file
+}
