--- a/soapui/src/main/java/com/eviware/soapui/analytics/AnalyticsHelper.java
+++ b/soapui/src/main/java/com/eviware/soapui/analytics/AnalyticsHelper.java
@@ -2,10 +2,6 @@
 
 import com.eviware.soapui.SoapUI;
 import com.eviware.soapui.Version;
-import com.eviware.soapui.analytics.providers.GoogleAnalyticsProviderFactory;
-import com.eviware.soapui.analytics.providers.LogTabAnalyticsProvider;
-import com.eviware.soapui.analytics.providers.OSUserProviderFactory;
-import com.eviware.soapui.analytics.providers.StatisticsCollectionConfirmationDialog;
 import com.eviware.soapui.model.settings.Settings;
 import com.eviware.soapui.settings.UISettings;
 
@@ -23,7 +19,7 @@
         Version optOutVersion = new Version(settings.getString(UISettings.ANALYTICS_OPT_OUT_VERSION, "0.0"));
         Version currentSoapUIVersion = new Version(SoapUI.SOAPUI_VERSION);
         if (!optOutVersion.getMajorVersion().equals(currentSoapUIVersion.getMajorVersion()) && SoapUI.usingGraphicalEnvironment()) {
-            disableAnalytics = StatisticsCollectionConfirmationDialog.showDialog() == JOptionPane.NO_OPTION;
+            disableAnalytics = true;
             settings.setBoolean(UISettings.DISABLE_ANALYTICS, disableAnalytics);
         }
         if (disableAnalytics) {
