--- a/soapui/src/main/java/com/eviware/soapui/analytics/AnalyticsHelper.java
+++ b/soapui/src/main/java/com/eviware/soapui/analytics/AnalyticsHelper.java
@@ -32,15 +32,11 @@
         if(isInitialize)
             return;
         isInitialize = true;
-        manager.setExecutorService(SoapUI.getThreadPool());
-        manager.registerAnalyticsProviderFactory(new OSUserProviderFactory());
         if (analyticsDisabled()) {
             return;
         }
 
-        manager.registerAnalyticsProviderFactory(new GoogleAnalyticsProviderFactory());
         if (System.getProperty("soapui.analytics.logtab", "false").equals("true")) {
-            manager.registerAnalyticsProviderFactory(new LogTabAnalyticsProvider.LogTabAnalyticsProviderFactory());
         }
     }
 }
