--- a/soapui/src/main/java/com/eviware/soapui/SoapUI.java
+++ b/soapui/src/main/java/com/eviware/soapui/SoapUI.java
@@ -19,7 +19,6 @@
 import com.eviware.soapui.actions.SaveAllProjectsAction;
 import com.eviware.soapui.actions.ShowSystemPropertiesAction;
 import com.eviware.soapui.actions.SoapUIPreferencesAction;
-import com.eviware.soapui.actions.SumbitUserInfoAction;
 import com.eviware.soapui.actions.SwitchDesktopPanelAction;
 import com.eviware.soapui.analytics.AnalyticsHelper;
 import com.eviware.soapui.impl.WorkspaceImpl;
@@ -888,8 +887,6 @@
 
         if (SoapUI.usingGraphicalEnvironment()) {
             if (workspace.isSupportInformationDialog()) {
-                SumbitUserInfoAction collector = new SumbitUserInfoAction();
-                collector.show();
                 workspace.setSupportInformationDialog(false);
             }
         }
