--- a/soapui/src/main/java/com/eviware/soapui/impl/wsdl/support/http/JCIFSEngine.java
+++ b/soapui/src/main/java/com/eviware/soapui/impl/wsdl/support/http/JCIFSEngine.java
@@ -20,6 +20,7 @@
 import jcifs.ntlmssp.Type1Message;
 import jcifs.ntlmssp.Type2Message;
 import jcifs.ntlmssp.Type3Message;
+import jcifs.ntlmssp.NtlmFlags;
 import jcifs.util.Base64;
 
 import org.apache.http.impl.auth.NTLMEngine;
@@ -40,7 +41,9 @@
         } catch (IOException ex) {
             throw new NTLMEngineException("Invalid Type2 message", ex);
         }
-        Type3Message t3m = new Type3Message(t2m, password, domain, username, workstation);
+        final int type2Flags = t2m.getFlags();
+        final int type3Flags = type2Flags & (0xffffffff ^ (NtlmFlags.NTLMSSP_TARGET_TYPE_DOMAIN | NtlmFlags.NTLMSSP_TARGET_TYPE_SERVER));
+        Type3Message t3m = new Type3Message(t2m, password, domain, username, workstation, type3Flags);
         return Base64.encode(t3m.toByteArray());
     }
 
