--- a/soapui/src/main/java/com/eviware/soapui/support/log/JettyLogger.java
+++ b/soapui/src/main/java/com/eviware/soapui/support/log/JettyLogger.java
@@ -104,4 +104,9 @@
     public void ignore(Throwable thrwbl) {
 
     }
+
+    @Override
+    public void debug(String string, long l) {
+        log.debug(string);
+    }
 }
--- a/soapui/src/main/java/com/eviware/soapui/monitor/SoapUIJettyThreadPool.java
+++ b/soapui/src/main/java/com/eviware/soapui/monitor/SoapUIJettyThreadPool.java
@@ -24,9 +24,8 @@
 
 public final class SoapUIJettyThreadPool implements ThreadPool {
     @Override
-    public boolean dispatch(Runnable arg0) {
+    public void execute(Runnable arg0) {
         SoapUI.getThreadPool().execute(arg0);
-        return true;
     }
 
     @Override
--- a/soapui/src/main/java/com/eviware/soapui/impl/wsdl/monitor/jettyproxy/JettyServer.java
+++ b/soapui/src/main/java/com/eviware/soapui/impl/wsdl/monitor/jettyproxy/JettyServer.java
@@ -31,7 +31,8 @@
 import java.net.InetSocketAddress;
 import java.net.Socket;
 import java.nio.channels.SocketChannel;
-import org.eclipse.jetty.server.AbstractHttpConnection;
+import org.eclipse.jetty.server.HttpChannel;
+import org.eclipse.jetty.server.HttpConnection;
 
 public class JettyServer extends org.eclipse.jetty.server.Server {
     private Logger log = Logger.getLogger(JettyServer.class);
@@ -45,7 +46,7 @@
     }
 
     @Override
-    public void handle(final AbstractHttpConnection connection) throws IOException, ServletException {
+    public void handle(final HttpChannel<?> connection) throws IOException, ServletException {
         final Request request = connection.getRequest();
 
         if (request.getMethod().equals("CONNECT")) {
@@ -70,9 +71,9 @@
             // response.setHeader("Connection", "close");
             response.flushBuffer();
 
-            IO.copyThread(socket.getInputStream(), out);
+            IO.copy(socket.getInputStream(), out);
 
-            IO.copyThread(in, socket.getOutputStream());
+            IO.copy(in, socket.getOutputStream());
         } else {
             super.handle(connection);
         }
--- a/soapui/src/main/java/com/eviware/soapui/impl/wsdl/monitor/SoapMonitorEngineImpl.java
+++ b/soapui/src/main/java/com/eviware/soapui/impl/wsdl/monitor/SoapMonitorEngineImpl.java
@@ -25,11 +25,14 @@
 import com.eviware.soapui.monitor.SoapUIJettyThreadPool;
 import com.eviware.soapui.support.StringUtils;
 import com.eviware.soapui.support.UISupport;
-import org.eclipse.jetty.server.bio.SocketConnector;
-import org.eclipse.jetty.server.ssl.SslSocketConnector;
+import org.eclipse.jetty.server.ServerConnector;
+import org.eclipse.jetty.util.ssl.SslContextFactory;
+import org.eclipse.jetty.server.SslConnectionFactory;
 import org.eclipse.jetty.servlet.ServletContextHandler;
 import org.eclipse.jetty.servlet.ServletHolder;
 
+import java.security.KeyStore;
+
 import java.net.BindException;
 
 public class SoapMonitorEngineImpl implements SoapMonitorEngine {
@@ -38,14 +41,15 @@
     private static final String HTTP = "http://";
     private static final String HTTPS = "https://";
     JettyServer server = new JettyServer();
-    SocketConnector connector = new SocketConnector();
-    private SslSocketConnector sslConnector;
+    ServerConnector connector;
+    private SslContextFactory sslConnector;
     private final String sslEndpoint;
     private boolean proxyOrTunnel = true;
     private ContentTypes includedContentTypes = SoapMonitorAction.defaultContentTypes();
 
     public SoapMonitorEngineImpl(final String sslEndpoint) {
         this.sslEndpoint = sslEndpoint;
+        this.connector = new ServerConnector(server);
     }
 
     public boolean isRunning() {
@@ -59,21 +63,19 @@
 
         if (!StringUtils.isNullOrEmpty(sslEndpoint)) {
             if (sslEndpoint.startsWith(HTTPS)) {
-                sslConnector = new SslSocketConnector();
-                sslConnector
-                        .setKeystore(settings.getString(SoapMonitorAction.SecurityTabForm.SSLTUNNEL_KEYSTORE, "JKS"));
-                sslConnector.setPassword(settings.getString(SoapMonitorAction.SecurityTabForm.SSLTUNNEL_PASSWORD, ""));
-                sslConnector.setKeyPassword(settings.getString(SoapMonitorAction.SecurityTabForm.SSLTUNNEL_KEYPASSWORD,
-                        ""));
-                sslConnector.setTruststore(settings.getString(SoapMonitorAction.SecurityTabForm.SSLTUNNEL_TRUSTSTORE,
-                        "JKS"));
-                sslConnector.setTrustPassword(settings.getString(
-                        SoapMonitorAction.SecurityTabForm.SSLTUNNEL_TRUSTSTORE_PASSWORD, ""));
+                sslConnector = new SslContextFactory();
+                sslConnector.setKeyStore(KeyStore.getInstance(settings.getString(SoapMonitorAction.SecurityTabForm.SSLTUNNEL_KEYSTORE, "JKS")));
+                sslConnector.setKeyStorePassword(settings.getString(SoapMonitorAction.SecurityTabForm.SSLTUNNEL_PASSWORD, ""));
+                sslConnector.setKeyManagerPassword(settings.getString(SoapMonitorAction.SecurityTabForm.SSLTUNNEL_KEYPASSWORD, ""));
+                sslConnector.setTrustStore(KeyStore.getInstance(settings.getString(SoapMonitorAction.SecurityTabForm.SSLTUNNEL_TRUSTSTORE, "JKS")));
+                sslConnector.setTrustStorePassword(settings.getString(SoapMonitorAction.SecurityTabForm.SSLTUNNEL_TRUSTSTORE_PASSWORD, ""));
                 sslConnector.setNeedClientAuth(false);
-                sslConnector.setMaxIdleTime(30000);
-                sslConnector.setPort(localPort);
 
-                server.addConnector(sslConnector);
+                //SslConnectionFactory scc = new SslConnectionFactory(contextFactory, "JKS"); 
+                //scc.setPort(localPort); 
+                //scc.setHost("localhost");
+
+                //server.addConnector(scc);
                 TunnelServlet tunnelServlet = new TunnelServlet(project, sslEndpoint, listenerCallBack);
                 tunnelServlet.setIncludedContentTypes(includedContentTypes);
                 context.addServlet(new ServletHolder(tunnelServlet), ROOT);
--- a/soapui/src/main/java/com/eviware/soapui/monitor/JettyMockEngine.java
+++ b/soapui/src/main/java/com/eviware/soapui/monitor/JettyMockEngine.java
@@ -34,17 +34,16 @@
 import org.eclipse.jetty.util.component.AbstractLifeCycle;
 import org.eclipse.jetty.io.Connection;
 import org.eclipse.jetty.io.EndPoint;
-import org.eclipse.jetty.io.nio.SelectChannelEndPoint;
+import org.eclipse.jetty.io.SelectChannelEndPoint;
 import org.eclipse.jetty.server.Connector;
-import org.eclipse.jetty.server.AsyncHttpConnection;
+import org.eclipse.jetty.server.HttpConnection;
 import org.eclipse.jetty.server.Request;
 import org.eclipse.jetty.server.RequestLog;
 import org.eclipse.jetty.server.Response;
 import org.eclipse.jetty.server.Server;
 import org.eclipse.jetty.server.handler.AbstractHandler;
 import org.eclipse.jetty.server.handler.RequestLogHandler;
-import org.eclipse.jetty.server.nio.SelectChannelConnector;
-import org.eclipse.jetty.server.ssl.SslSocketConnector;
+import org.eclipse.jetty.server.ServerConnector;
 
 import javax.servlet.ServletException;
 import javax.servlet.ServletInputStream;
@@ -66,8 +65,6 @@
 import java.util.Map;
 import java.util.Set;
 import java.util.concurrent.CopyOnWriteArrayList;
-import org.eclipse.jetty.io.AsyncEndPoint;
-import org.eclipse.jetty.io.nio.AsyncConnection;
 
 /**
  * Core Mock-Engine hosting a Jetty web server
@@ -83,7 +80,7 @@
     private Map<Integer, SoapUIConnector> connectors = new HashMap<Integer, SoapUIConnector>();
     private List<MockRunner> mockRunners = new CopyOnWriteArrayList<MockRunner>();
 
-    private SslSocketConnector sslConnector;
+    private SslConnection sslConnector;
 
     private boolean addedSslConnector;
 
@@ -182,7 +179,7 @@
         logHandler.setRequestLog(new MockRequestLog());
         server.addLifeCycle(logHandler);
 
-        sslConnector = new SslSocketConnector();
+        sslConnector = new SslConnection();
         sslConnector.setMaxIdleTime(30000);
     }
 
@@ -254,7 +251,7 @@
         }
     }
 
-    private class SoapUIConnector extends SelectChannelConnector {
+    private class SoapUIConnector extends ServerConnector {
         private Set<Connection> connections = new HashSet<Connection>();
 
         @Override
@@ -270,7 +267,7 @@
         }
 
         @Override
-        public AsyncConnection newConnection(SocketChannel socketChannel, AsyncEndPoint selectChannelEndPoint) {
+        public AsyncConnection newConnection(SocketChannel socketChannel, EndPoint selectChannelEndPoint) {
             return new SoapUIHttpConnection(SoapUIConnector.this, selectChannelEndPoint, getServer());
         }
 
@@ -295,7 +292,7 @@
         }
     }
 
-    private class SoapUIHttpConnection extends AsyncHttpConnection {
+    private class SoapUIHttpConnection extends HttpConnection {
         private CapturingServletInputStream capturingServletInputStream;
         private BufferedServletInputStream bufferedServletInputStream;
         private CapturingServletOutputStream capturingServletOutputStream;
