--- a/soapui/src/main/java/com/eviware/soapui/impl/wsdl/support/http/ProxyVoleUtil.java
+++ b/soapui/src/main/java/com/eviware/soapui/impl/wsdl/support/http/ProxyVoleUtil.java
@@ -15,8 +15,8 @@
 */
 package com.eviware.soapui.impl.wsdl.support.http;
 
-import com.btr.proxy.search.ProxySearch;
-import com.btr.proxy.util.PlatformUtil;
+import com.github.markusbernhardt.proxy.ProxySearch;
+import com.github.markusbernhardt.proxy.util.PlatformUtil;
 
 /**
  * @author Joel
--- a/soapui/src/main/java/com/eviware/soapui/impl/wsdl/support/http/SchemeProxyFilter.java
+++ b/soapui/src/main/java/com/eviware/soapui/impl/wsdl/support/http/SchemeProxyFilter.java
@@ -15,7 +15,7 @@
 */
 package com.eviware.soapui.impl.wsdl.support.http;
 
-import com.btr.proxy.util.UriFilter;
+import com.github.markusbernhardt.proxy.util.UriFilter;
 
 import java.net.URI;
 
--- a/soapui/src/main/java/com/eviware/soapui/impl/wsdl/support/http/ProxyUtils.java
+++ b/soapui/src/main/java/com/eviware/soapui/impl/wsdl/support/http/ProxyUtils.java
@@ -16,8 +16,8 @@
 
 package com.eviware.soapui.impl.wsdl.support.http;
 
-import com.btr.proxy.selector.whitelist.ProxyBypassListSelector;
-import com.btr.proxy.util.UriFilter;
+import com.github.markusbernhardt.proxy.selector.whitelist.ProxyBypassListSelector;
+import com.github.markusbernhardt.proxy.util.UriFilter;
 import com.eviware.soapui.SoapUI;
 import com.eviware.soapui.model.propertyexpansion.PropertyExpander;
 import com.eviware.soapui.model.propertyexpansion.PropertyExpansionContext;
